docs/source/citation_registry.rst

```diff
@@ -0,0 +1,73 @@
+.. _cite_as:
+
+:orphan:
+
+Landlab Citation Registry
+-------------------------
+
+A relatively new interface also automates the process of extracting citations
+for Landlab components. Components with required software citations have an
+attribute called ``cite_as``.
+
+.. code-block:: python
+
+    from landlab.components import OverlandFlow
+    OverlandFlow.cite_as
+
+This will give::
+
+    @article{adams2017landlab,
+           title={The Landlab v1.0 OverlandFlow component: a Python
+                  tool for computing shallow-water flow across watersheds},
+                  author={Adams, Jordan M and Gasparini, Nicole M and
+                  Hobley, Daniel EJ and Tucker, Gregory E and
+                  Hutton, Eric WH and Nudurupati, Sai S and
+                  Istanbulluoglu, Erkan},
+                  journal={Geoscientific Model Development},
+                  volume={10},
+                  number={4},
+                  pages={1645},
+                  year={2017},
+                  publisher={Copernicus GmbH}
+                  }
+
+In addition, you can use the "citation registry" to make a .bib file of all
+citations you've used in script.
+
+.. code-block:: python
+
+    import landlab
```

Comment:
In this branch, this works for me in 
```python
import landlab
w = landlab.registry.format_citations()
with open("citations.bib", "w") as f:
    f.write(w)
```
Although, it breaks with a component, e.g.:
```python
import landlab
from landlab.components import LinearDiffuser

mg = landlab.RasterModelGrid((3,3))
mg.add_ones('topographic__elevation', at='node')
ed = LinearDiffuser(mg)

w = landlab.registry.format_citations()
with open("citations.bib", "w") as f:
    f.write(w)
```
I get this error
```python
  File "/Users/njlyons/repos/landlab/landlab/landlab/core/messages.py", line 139, in indent_and_wrap
    first_line, the_rest = [lines[0].strip()], lines[1:]

IndexError: list index out of range
```

Reply:
@nathanlyons I've addressed this by removing these examples and replacing them with the docstrings for the `landlab._registry` file. These were not represented in the docs... and are actually tested. I think that is a better solution. 

Thanks for testing these!